static/js/components/search/FinalGradeRangeAccessor.js

```diff
@@ -0,0 +1,78 @@
+import * as React from "react";
+import {
+  RangeFilter, RangeAccessor, RangeQuery,
+  BoolMust, HistogramBucket, FilterBucket,
+  NestedBucket, NestedQuery, TermQuery
+} from "searchkit";
+import _ from 'lodash';
+
+export default class FinalGradeRangeAccessor extends RangeAccessor {
+
+  buildSharedQuery(query) {
+    if (this.state.hasValue()) {
+      let val: any = this.state.getValue();
+      let rangeFilter = this.fieldContext.wrapFilter(RangeQuery(this.options.field, {
+        gte: val.min, lte: val.max
+      }));
+      let selectedFilter = {
+        name: this.translate(this.options.title),
+        value: `${val.min} - ${val.max}`,
+        id: this.options.id,
+        remove: ()=> {
+          this.state = this.state.clear()
+        }
+      };
+      let nestedQuery = NestedQuery('program.final_grades', rangeFilter);
+
+      query = query
+        .addFilter(this.key, nestedQuery)
+        .addSelectedFilter(selectedFilter);
+      return query;
+    }
+
+    return query
+  }
+
+  buildOwnQuery(query) {
```

Comment:
I think it would probably be helpful to put some code comments in this method about what we're doing here. I think I get the general idea, but it might be easier if we ever have to revisit this code (and given SearchKit's propensity to breaking things that's probably a certainty :roll_eyes: ).